numpy/lib/arraysetops.py

```diff
@@ -590,18 +590,25 @@ def in1d(ar1, ar2, assume_unique=False, invert=False):
     ar1 = np.asarray(ar1).ravel()
     ar2 = np.asarray(ar2).ravel()
 
-    # Ensure that iteration through object arrays yields size-1 arrays
-    if ar2.dtype == object:
-        ar2 = ar2.reshape(-1, 1)
-
     # Check if one of the arrays may contain arbitrary objects
     contains_object = ar1.dtype.hasobject or ar2.dtype.hasobject
 
+    # String is counted as arbitary python object above however it can be sorted 
+    if(all(map(lambda i: isinstance(i, str), ar1)) 
+        and all(map(lambda i: isinstance(i, str), ar2))):
+        contains_object = False
```

Comment:
This manual iteration here is an anti-pattern in my opinion.  As is it even makes everything else (and not just objects) slower.

Can you rather find out why `contains_object` rejects the alternative path?  Is it just historical?